core/src/main/java/org/jdbi/v3/core/argument/EnumArgumentFactory.java

```diff
@@ -49,7 +52,14 @@
     }
 
     private static <E extends Enum<E>> Optional<Argument> byName(E value, ConfigRegistry config) {
-        return makeArgument(Types.VARCHAR, String.class, value, E::name, config);
+        final Function<E, String> transform = e -> {
```

Comment:
Not what I meant :) This is best suited to a private static method, maybe `String enumToName(E e)`, used by `makeArgument` as `EnumArgumentFactory::enumToName`

Reply:
I prefer to keep the enum-to-name conversion in the in the `byName` method as is it not used anywhere else. This way it’s not needed to look for the `enumToName` method when trying to understand what `byName` does and one does not have to look for other uses of the `enumToName` method when it’s required to modify it somehow. Not that hard (especially using an IDE), but would still require some effort.

Reply:
TLDR: I'll not argue this further after this post so that this feature doesn't get hung up on nitpicks, but I would like another reviewer to cut the knot and either accept the multi-line pure lambda or request a method.

--

I disagree with some of your concepts. `as is it not used anywhere else` is what `private` methods are for. `not needed to look for the enumToName method`, sure, only for the `enumToName` variable instead, which is really no difference as both blocks are code are declared one or two lines away. `look for other uses` really not a concern with `private` members of such a small class.

My main objection is just that multi-line lambdas that don't rely on external state (pure functions) have no business being lambdas. A lambda declared as a variable inside a method (different from anonymous lambdas used with streams) suggests the lambda's content relies on some state of the declaring method. If they don't (such as here), what you're really doing is writing a method inside of another method and falsely suggesting it has a mutable nature, which is just confusing.

Reply:
I did end up flattening this method out, by using `Optional` the logic became clearer.